common/component/sql/migrations/postgres/postgres_migrations.go

```diff
@@ -37,33 +37,58 @@ type Migrations struct {
 
 // Perform the required migrations
 func (m Migrations) Perform(ctx context.Context, migrationFns []commonsql.MigrationFn) error {
-	// Use an advisory lock (with an arbitrary number) to ensure that no one else is performing migrations at the same time
-	// This is the only way to also ensure we are not running multiple "CREATE TABLE IF NOT EXISTS" at the exact same time
-	// See: https://www.postgresql.org/message-id/CA+TgmoZAdYVtwBfp1FL2sMZbiHCWT4UPrzRLNnX1Nb30Ku3-gg@mail.gmail.com
-	const lockID = 42
+	// Ensure the metadata table exists
```

Comment:
Reviewers: I had to change how we lock before migrations because advisory locks aren't available in CockroachDB. This is functionally equivalent (and the certification tests are validating it), but because we call EnsureMetadataTable regardless of the lock, that operation may fail and may need to be retried automatically (see implementation for EnsureMetadataTable)